### ISSUE TITLE
Beta: Actor not automatically dropped when last strong Address is dropped

### ISSUE THREAD

**Issue by seijikun** [Restioson/xtra#25]
In the current beta `0.5.0-beta.5`, an actor is not automatically dropped when its last `Address` is dropped.
This behavior can be caused by the following code:

```
use xtra::prelude::*;
use xtra::spawn::AsyncStd;

struct Printer {
    times: usize,
}

impl Printer {
    fn new() -> Self {
        Printer { times: 0 }
    }
}

#[async_trait::async_trait]
impl Actor for Printer {
    async fn stopped(&mut self) {
        println!("Printer is stopped");
    }
}

struct Print(String);
impl Message for Print {
    type Result = ();
}

#[async_trait::async_trait]
impl Handler<Print> for Printer {
    async fn handle(&mut self, print: Print, _ctx: &mut Context<Self>) {
        self.times += 1;
        println!("Printing {}. Printed {} times so far.", print.0, self.times);
    }
}

#[async_std::main]
async fn main() {
    {
        let addr = Printer::new().create(None).spawn(&mut AsyncStd);
        addr.send(Print("hello".to_string()))
            .await
            .expect("Printer should not be dropped");
    }
    println!("Printer should be dropped here");
}

```
With this code, Printer is actually never dropped, even until the application stops.

This is probably because of [refcount.rs line 74](https://github.com/Restioson/xtra/blob/e2570f273513f3404edc25d3a4284bb6023e6ea5/src/refcount.rs#L74), where the strong refcount is checked against `==2`.
It's only `1` in the example, though.

**Comment by Restioson** [comment]
Good catch! Looks like I missed this when I changed context to having a weak refcount rather than a strong one.

**Comment by Restioson** [comment]
This has been fixed for a while, I just forgot to close the issue :sweat_smile: Fixed in 423ba56c00b1ecd82123b18ef4711e9d93b1419b
